app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java

```diff
@@ -0,0 +1,96 @@
+package com.getsensibill.spendmanager.demo.fragment;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.FragmentTransaction;
+
+import android.os.Bundle;
+
+import com.getsensibill.spendmanager.demo.R;
+import com.getsensibill.spendmanager.demo.databinding.ActivityFragmentJavaBinding;
+import com.getsensibill.web.data.UiFinishReason;
+import com.getsensibill.web.data.configuration.NavigationIntent;
+import com.getsensibill.web.data.configuration.ProgrammaticTheme;
+import com.getsensibill.web.data.models.Brand;
+import com.getsensibill.web.ui.WebUiFragment;
+import com.getsensibill.web.ui.WebUiNetworkErrorFragment;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Directly using the [WebUiFragment]
+ * <p>
+ * When using [WebUiFragment] directly, please ensure to properly implement the [Listener] methods.
+ * Also, please use, or see [WebUiNetworkErrorFragment] as reference for what to display when
+ * [Listener.onDisplayNetworkError] is called.
+ */
+public class FragmentJavaActivity extends AppCompatActivity implements WebUiFragment.Listener {
+    private ActivityFragmentJavaBinding binding;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        binding = ActivityFragmentJavaBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        if(savedInstanceState == null) {
+            WebUiFragment fragment = new WebUiFragment();
+            Bundle bundle = fragment.getArguments();
+            if (bundle == null) {
```

Comment:
If we're always creating a new fragment above, `.getArguments()` will always be null, so we can just say `Bundle bundle = new Bundle();`

Reply:
updated